hudi-spark-datasource/hudi-spark-common/src/main/java/org/apache/hudi/DataSourceUtils.java

```diff
@@ -227,9 +231,21 @@ public static HoodieWriteResult doWriteOperation(SparkRDDWriteClient client, Jav
     }
   }
 
-  public static HoodieWriteResult doDeleteOperation(SparkRDDWriteClient client, JavaRDD<HoodieKey> hoodieKeys,
-      String instantTime) {
-    return new HoodieWriteResult(client.delete(hoodieKeys, instantTime));
+  public static HoodieWriteResult doDeleteOperation(SparkRDDWriteClient client, JavaRDD<Tuple2<HoodieKey, scala.Option<HoodieRecordLocation>>> hoodieKeysAndLocations,
+      String instantTime, boolean isPrepped) {
+
+    if (isPrepped) {
+      JavaRDD<HoodieRecord> records = hoodieKeysAndLocations.map(tuple -> {
+        HoodieRecord record = client.getConfig().getRecordMerger().getRecordType() == HoodieRecord.HoodieRecordType.AVRO
```

Comment:
"client" object is in driver. since here we are accessing it in the executor, we might get NotSerializableException.

Reply:
can you declare the record type as a variable outside and then just use that variable.

Reply:
Fixed. Moved client outside of the `map` function.